### PR TITLE
chore: updated copy functionality to include element imports

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description

`extractNodeCustomElements` method which runs over the copied node text and finds out the custom lit elements starting with `sp-`
`generateImportStatements` returns the created import statements of the elements which returns
```js
const fullCopiedText = `${importStatements}\n${node.textContent}`;
```

## Related issue(s)

- https://github.com/adobe/spectrum-web-components/issues/2291
- https://github.com/adobe/spectrum-web-components/issues/3517

## How has this been tested?

-   [x] Copy to clipboard of code snippets with dependencies
    1. Go [here](https://docs-copy-paste--spectrum-web-components.netlify.app/components/accordion/#example)
    2. Click on "Copy to Clipboard" button
    3. Paste it on any notepad
    4. See how the dependencies are imported with the code snippet
-   [ ] Copy to clipboard for only dependencies
    1. Go [here](https://docs-copy-paste--spectrum-web-components.netlify.app/components/accordion/#usage)
    2. Click on "Copy to Clipboard" button
    3. Paste it on any notepad
    4. See how the dependencies are not imported with the line

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by blunteshwar** [comment]
Can you either remove or fill the `How has this been tested?` section.
Otherwise LGTM, once it is updated to main i think we can merge it.

**Comment by nikkimk** [comment]
@Rajdeepc I like @blunteshwar's recommendation to fill or remove the `How has this been tested?` section. Consider this my +1 to the comment.

**Comment by Rajdeepc** [comment]
> @Rajdeepc I like @blunteshwar's recommendation to fill or remove the `How has this been tested?` section. Consider this my +1 to the comment.

yea my bad! When a PR is opened the preview docs are not surfaced. It should be there now
cc @blunteshwar

**Review by TarunAdobe — DISMISSED** [review]

File: projects/documentation/src/components/copy-to-clipboard.ts
Comment (TarunAdobe) [reply projects/documentation/src/components/copy-to-clipboard.ts:64]: I love this idea. Let's make sure we always write a documentation block above all the new and important functions we introduce in the library now onwards 🥇

Comment (Rajdeepc) [reply projects/documentation/src/components/copy-to-clipboard.ts:64]: Good catch!

**Pull request merged**
